src/Aspire.Hosting.LocalStack/LocalStackResourceBuilderExtensions.cs

```
@@ -177,6 +179,31 @@ public static IDistributedApplicationBuilder UseLocalStack(this IDistributedAppl
             resourceBuilder = resourceBuilder.WithEnvironment(key, value);
         }
 
+        if (containerOptions.EagerLoadedServices.Count == 0)
+        {
+            resourceBuilder = resourceBuilder.WithHttpHealthCheck("/_localstack/health", 200, LocalStackResource.PrimaryEndpointName);
+        }
+        else
+        {
+            resourceBuilder = resourceBuilder.WithEnvironment("EAGER_SERVICE_LOADING", "1");
```

I think setting the env variables here is reasonable. That said, there’s a subtle edge case:

Inside `AddLocalStack` we already have:

```csharp
// Add any additional environment variables
foreach (var (key, value) in containerOptions.AdditionalEnvironmentVariables)
{
    resourceBuilder = resourceBuilder.WithEnvironment(key, value);
}
```

If the same env variables are passed in both places, I’m not sure which one would win, probably the ones added last, but it’s not obvious.

I don’t have a perfect solution right now, but a couple of future options could be:

- Dropping `AdditionalEnvironmentVariables` entirely and consolidating everything into containerOptions.
- Adding a validation step to detect overlapping variables and warn/fail early.

The first approach might be too restrictive, so I’m leaning toward the validation idea as a safer middle ground.

Either way, I don’t think we need to solve this in this PR, I’m not expecting that. I just wanted to share this thought for future consideration.